runtime/jvmti/jvmtiCapability.c

```diff
@@ -478,8 +478,13 @@ jvmtiRelinquishCapabilities(jvmtiEnv* env,
 
 #if JAVA_SPEC_VERSION >= 11
 		if (capabilities_ptr->can_generate_sampled_object_alloc_events) {
-			/* The default sampling interval is not changed. */
+			J9VMThread *currentThread = NULL;
 			jvmtiData->flags &= ~J9JVMTI_FLAG_SAMPLED_OBJECT_ALLOC_ENABLED;
+			rc = getCurrentVMThread(vm, &currentThread);
+			if (JVMTI_ERROR_NONE == rc) {
+				/* Set sampling interval to UDATA_MAX to inform GC that sampling is not required */
+				vm->memoryManagerFunctions->j9gc_set_allocation_sampling_interval(currentThread, UDATA_MAX);
```

Comment:
If the sampling interval is global, the GC call should be changed to take a VM, not a thread.

Reply:
@dmitripivkine Any objection?

Reply:
Seems right for me. Is there any thread-specific usage of this API I missed?

Reply:
> Is there any thread-specific usage of this API I missed?

No, the `currentThread` is used to retrieve `J9JavaVM` instead.

Reply:
> Seems right for me. Is there any thread-specific usage of this API I missed?

we need MM_EnvironmentBase to retrieve _objectAllocationInterface for disableCachedAllocations() in https://github.com/eclipse/openj9/pull/9254

Reply:
> we need MM_EnvironmentBase to retrieve _objectAllocationInterface for disableCachedAllocations() in #9254

Post a question at https://github.com/eclipse/openj9/pull/9254/files#r409765714

Reply:
Maybe we can retrieve MM_EnvironmentBase via getCurrentVMThread(), or somewhere else

Reply:
There is not a question how to get thread info, if we really need it we can keep API as it stays now. However there is a question about correctness global vs per thread. Is it correct to set is as now for thread or we should manage to set this for ALL threads?

Reply:
My complaint here was all the work to retrieve the thread, only to have it used to fetch the VM.

Reply:
`j9gc_set_allocation_sampling_interval()` is intended to affect all threads. If https://github.com/eclipse/openj9/pull/9254 can find a way for `disableCachedAllocations()` via `J9JavaVM *` instead of `J9VMThread *`, `j9gc_set_allocation_sampling_interval()` can be modified to take `J9JavaVM *` which gives clear indication that this API is for VM all threads.

Reply:
I believe this is true. However current implementation in https://github.com/eclipse/openj9/pull/9254 sets flag for one thread only. It does not look correct for me. We need to double check. And it confirms GAC position - it is useful to set correct scope

Reply:
Okay, am going to change `j9gc_set_allocation_sampling_interval()` to take `J9JavaVM *`.

@LinHu2016 please adjust your PR & personal build.

Reply:
sure, I am going to handle whole threads for disableCacheAllocations and monitor extensions->disableInlineAllocationForSamplingBytesGranularity for initializing _cachedAllocationsEnabled for new created thread, I am going to leave j9gc_set_allocation_sampling_interval() no change until this PR is merged.

Reply:
Updated the PR to accept `J9JavaVM *` in `j9gc_set_allocation_sampling_interval()`. Keep the `NULL` check for `currentThread` within `disposeEnvironment()`. `deleteAgentBreakpoint(J9VMThread *, J9JVMTIEnv *, J9JVMTIAgentBreakpoint *)` and its associates might not need `J9VMThread *currentThread`, but will leave that to another PR.

@gacholio @DanHeidinga please review.